src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -224,6 +225,40 @@ public TaskData getTaskByID(long taskID) {
     return ancestors;
   }
 
+  public ArrayList<TaskTreeData> getTaskTree(long projectID) {
+    ArrayList<TaskTreeData> taskTree = new ArrayList<>();
+    Filter filter = new FilterPredicate("projectID", FilterOperator.EQUAL, projectID);
+    ArrayList<TaskData> tasks = getTasks(filter, NO_QUERY_LIMIT, NO_QUERY_SORT);
+    Iterator<TaskData> iterator = tasks.iterator();
+    while (iterator.hasNext()) {
+      TaskData task = iterator.next();
+      if (task.getParentTaskID() == 0) {
+        taskTree.add(new TaskTreeData(task));
+        iterator.remove();
+      }
+    }
+    buildTree(taskTree, tasks);
+    return taskTree;
+  }
+
+  private void buildTree(ArrayList<TaskTreeData> taskTree, ArrayList<TaskData> tasks) {
+    if (tasks.isEmpty()) {
+      return;
+    }
+    ArrayList<TaskTreeData> nextTreeLayer = new ArrayList<>();
+    for (TaskTreeData taskTreeNode : taskTree) {
+      Iterator<TaskData> iterator = tasks.iterator();
+      while (iterator.hasNext()) {
+        TaskData task = iterator.next();
+        if (task.getParentTaskID() == taskTreeNode.getTask().getTaskID()) {
+          nextTreeLayer.add(taskTreeNode.addSubtask(task));
+          iterator.remove();
```

Comment:
Repeatedly removing items from the list could get expensive since it'll need to shift all items on the right one step to the left.  Consider building your tree without having to shift the list repeatedly.